src/extends/room/movement.js

```diff
@@ -83,14 +85,22 @@ Room.getStructuresCostmatrix = function (roomname, opts) {
     const room = Game.rooms[roomname]
     const structures = room.find(FIND_STRUCTURES)
     for (let structure of structures) {
-      if (!opts.ignoreRoads && structure.structureType === STRUCTURE_ROAD) {
+      if (structure.structureType === STRUCTURE_ROAD) {
+        if (!opts.ignoreRoads) {
```

Comment:
The `!` needs to be removed from this if condition (and the same on new line 96, when checking if portals are to be ignored).